### DISCUSSION TITLE
Converting ExpandoObject to PyObject

### DISCUSSION THREAD

**Discussion by jaqueamo** [pythonnet/pythonnet#1719] [Q&A]
Hello :)

I want to call a C# method from Python returning a ExpandoObject - properties are added dynamic during runtime. 
In Python I want to access those properties like object.property (this is possible in C# with the dynamic keyword and ExpandoObject).

Therfore I wrote a Codec which adds the properties to a PyObject:

```csharp
using System;
using System.Collections.Generic;
using System.Dynamic;

using Python.Runtime;

namespace Python
{
    public class ExpandoObjectEncoder : IPyObjectEncoder
    {
        public bool CanEncode(Type type)
        {
            return type == typeof(ExpandoObject);
        }

        public PyObject TryEncode(object value)
        {
            if (value is not ExpandoObject expandoObject)
            {
                return null;
            }

            PyObject obj = PyObject.FromManagedObject(new object());
            IDictionary<string, object> expandoDictionary = expandoObject;

            foreach (KeyValuePair<string, object> keyValuePair in expandoDictionary)
            {
                obj.SetAttr(keyValuePair.Key, keyValuePair.Value.ToPython());
            }

            return obj;
        }
    }
}
```

With this codec I can access the properties in Python like I want to.
 
I Just want to know if this good practice or if there are other ways to achieve this ? 

Thank you.

**Comment by lostmsu** [comment]
If your solution works for you - great. As far as I can see, you won't be able to set properties of the object encoded this way though.

There is a [feature request for the full support of DLR](https://github.com/pythonnet/pythonnet/issues/72) (which would include `ExpandoObject`), that I personally had a few attempts at, but it is not a small feat, and us project maintainers do not plan implementing it.
